torch/jit/_recursive.py

```diff
@@ -540,6 +540,10 @@ def check_module_initialized(mod):
     if not hasattr(mod, '_parameters'):
         raise RuntimeError("'{}' has not been initialized, did you forget to call 'super()'?"
                            .format(torch.typename(type(mod))))
+    for param in mod.parameters():
```

Comment:
so the test failures can be seen in the full error message: https://circleci.com/api/v1.1/project/github/pytorch/pytorch/8056474/output/105/0?file=true&allocation-id=5f7d8d4f7099d65d3111d1e5-0-build%2F48B29293

I think it comes from this line:
```
Oct 07 11:29:49 Traceback (most recent call last):
Oct 07 11:29:49   File "/Users/distiller/workspace/miniconda3/lib/python3.7/site-packages/torch/testing/_internal/common_distributed.py", line 259, in wrapper
Oct 07 11:29:49     fn()
Oct 07 11:29:49   File "/Users/distiller/workspace/miniconda3/lib/python3.7/site-packages/torch/testing/_internal/dist_utils.py", line 90, in new_test_method
Oct 07 11:29:49     return_value = old_test_method(self, *arg, **kwargs)
Oct 07 11:29:49   File "/Users/distiller/workspace/miniconda3/lib/python3.7/site-packages/torch/testing/_internal/distributed/nn/api/remote_module_test.py", line 132, in test_forward_async
Oct 07 11:29:49     for remote_module in self._create_remote_module_iter(dst_worker_name):
Oct 07 11:29:49   File "/Users/distiller/workspace/miniconda3/lib/python3.7/site-packages/torch/testing/_internal/distributed/nn/api/remote_module_test.py", line 103, in _create_remote_module_iter
Oct 07 11:29:49     scripted_remote_module = torch.jit.script(remote_module)
Oct 07 11:29:49   File "/Users/distiller/workspace/miniconda3/lib/python3.7/site-packages/torch/jit/_script.py", line 898, in script
Oct 07 11:29:49     obj, torch.jit._recursive.infer_methods_to_compile
Oct 07 11:29:49   File "/Users/distiller/workspace/miniconda3/lib/python3.7/site-packages/torch/jit/_recursive.py", line 350, in create_script_module
Oct 07 11:29:49     check_module_initialized(nn_module)
Oct 07 11:29:49   File "/Users/distiller/workspace/miniconda3/lib/python3.7/site-packages/torch/jit/_recursive.py", line 543, in check_module_initialized
Oct 07 11:29:49     for param in mod.parameters():
Oct 07 11:29:49   File "/Users/distiller/workspace/miniconda3/lib/python3.7/site-packages/torch/distributed/nn/api/remote_module.py", line 266, in parameters
Oct 07 11:29:49     "Method ``parameters`` not supported for RemoteModule. Please use ``remote_parameters`` instead."
Oct 07 11:29:49 ValueError: Method ``parameters`` not supported for RemoteModule. Please use ``remote_parameters`` instead.
```

I'm not sure what's the right condition to put here - maybe check that it's not RemoteModule? @suo or @mrshenli  probably know

Reply:
Note that this is failing internally as well because we call this module with things that are "almost" nn.Module but don't have a `.paramters()` method.
Maybe you should check if there is a `._parameters` dict and if so traverse it?

Reply:
Thanks, I will fix this.